flare/gp.py

```diff
@@ -119,22 +121,39 @@ def force_list_to_np(forces: list):
 
         return forces_np
 
-    def train(self, output=None, custom_bounds=None,
-              grad_tol: float = 1e-4,
-              x_tol: float = 1e-5,
-              line_steps: int = 20):
-        """Train Gaussian Process model on training data. Tunes the \
+    def train(self, output=None, opt_param_override: dict = None):
+        """
+        Train Gaussian Process model on training data. Tunes the \
 hyperparameters to maximize the likelihood, then computes L and alpha \
-(related to the covariance matrix of the training set)."""
+(related to the covariance matrix of the training set).
 
+        :param output: Output to write to
+        :param opt_param_override: Dictionary of parameters to override
+        instace's optimzation parameters.

```

Comment:
This could be a fun place to try out one of Python 3.8's new features! [TypedDicts](https://www.python.org/dev/peps/pep-0589/)! I actually don't know if this would be a good use case, but if we turned the `opt_param_override` into a TypedDict then Python's type checker could check the types in the dictionary

I am just looking for use cases of the new features of python so totally unnecessary to actually implement...